radicle-node/src/service.rs

```diff
@@ -943,14 +947,49 @@ where
         }
     }
 
-    fn maintain_connections(&mut self) {
-        // TODO: Connect to all potential seeds.
-        if self.sessions.len() < TARGET_OUTBOUND_PEERS {
-            let delta = TARGET_OUTBOUND_PEERS - self.sessions.len();
-
-            for _ in 0..delta {
-                // TODO: Connect to random peer.
+    fn choose_addresses(&mut self) -> Vec<Address> {
+        let mut initializing: Vec<Address> = Vec::new();
+        let mut negotiated: HashMap<NodeId, &Session> = HashMap::new();
+        for s in self.sessions.values() {
+            if !s.link.is_outbound() {
```

Comment:
why is 's.link.is_outbound()' preferred over 's.link == outbound'?

Reply:
because it doesn't require the `Link` type to be in scope :shrug: